### PR TITLE
Simplify handlers (internal only)

### PR DESCRIPTION
**Pull request by rougeth**
This pull request make easier to create or extend handlers. Now Bottery choose the view to respond a new message according to a list of handler instances.

Closes: https://github.com/rougeth/bottery/issues/161

### CODE REVIEW

File: docs/handlers.rst [hunk 1]
```diff
@@ -19,22 +19,23 @@ If the bot receives a message that is exactly equal to `ping` the response will
 
 Bottery has the following handlers available:
 
+
```
Comment (leportella) [inline docs/handlers.rst]: why two lines?

File: docs/handlers.rst [hunk 2]
```diff
@@ -114,3 +117,26 @@ If you want to have a default handler, executed if no message is previously capt
         # previously defined handlers matches the message
         handlers.default(default_response),
     ]
+
+
+Create a new handler
+--------------------
+
+The interface of a handler is quite simple. To create a new one, everything needed is a class with a method called ``match`` and an attribute called ``view``. The method ``match`` define if that handler should be used or not to respond to the new message. It must receive a ``Message`` as a parameter and return a boolean. The attribute ``view`` will be used if the ``match`` returns ``True``.
```
Comment (leportella) [inline docs/handlers.rst]: I think it is better to use "the handler's structure" instead of interface

File: docs/handlers.rst [hunk 3]
```diff
@@ -114,3 +117,26 @@ If you want to have a default handler, executed if no message is previously capt
...
+
+In the example below, there is a handler that checks if message has more than 50 words:
+
+.. code-block:: python
+
+    class LongTextHandler:
```
Comment (leportella) [inline docs/handlers.rst]: I think i would be clearer if the example had a pattern to compare, like `StartsWithHandler` or something similar. I got confused if this did the same thing.

File: docs/handlers.rst (hunk 1 above)
Comment (rougeth) [reply docs/handlers.rst]: Two lines between sections. I changed it a bit, is it better?

**Pull request merged**